/**
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
syntax = "proto3";

package logproto;

import "google/protobuf/timestamp.proto";

service Pusher {
  rpc Push(PushRequest) returns (PushResponse) {}
}

message PushRequest {
  repeated StreamAdapter streams = 1;
}

message PushResponse {}

message StreamAdapter {
  string labels = 1;
  repeated EntryAdapter entries = 2;
  // hash contains the original hash of the stream.
  uint64 hash = 3;
}

message LabelPairAdapter {
  string name = 1;
  string value = 2;
}

message EntryAdapter {
  google.protobuf.Timestamp timestamp = 1;
  string line = 2;
  repeated LabelPairAdapter nonIndexedLabels = 3;
}
